### ISSUE TITLE
Request - Could `Step` have a `getPolicy` method to get the associated policy?

### ISSUE THREAD

**Issue by valentine195** [apigee/apigeelint#525]
In some of our custom rules, we are requiring certain policies to be present in proxy endpoint preflow request phases. This requires getting the list of steps and looking up the corresponding policy out of the bundle. It would be nice if `Step` had a `getPolicy()` method to return the associated policy out of the bundle list.

Something like this:

```js
Step.prototype.getPolicy = function () {
  if (!this.policy) {
    //not sure this is the best way to find the bundle but it works ¯\_(ツ)_/¯
    let parent = this.parent;
    while (parent && !parent.getPolicyByName) {
      parent = parent.getParent();
    }
    this.policy = parent.getPolicyByName(this.getName());
  }
  return this.policy;
}
```

Correspondingly, `Policy.getSteps()` could associate as well:

```js
this.steps.forEach(step => step.policy = this);
```

**Comment by DinoChiesa** [comment]
Seems like a good suggestion!

**Comment by valentine195** [comment]
> If I am understanding it correctly, you could also navigate the other way. Get the ProxyEndpoints, then get the Preflow Request , and get the Steps there...

Yeah, this is what I was currently doing, this would just be a QoL improvement. Getting the Preflow Request actually doesn't currently work for me due to #521 but I do essentially the same thing.
